plugins/inputs/timex/timex.go

```diff
@@ -0,0 +1,102 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package timex
+
+import (
+	_ "embed"
+	"fmt"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	// https://man7.org/linux/man-pages/man2/adjtimex.2.html#NOTES.
+	// There the frequency is represented as a fixed-point number with a scaling factor of 2^16 (65536).
+	ppm16 = float64(65536)
+)
+
+type Timex struct{}
```

Comment:
Missing godoc comment
```suggestion
// Timex gathers system time metrics using the Linux kernel adjtimex syscall.
type Timex struct{}
```